src/write/line.rs

```diff
@@ -0,0 +1,1517 @@
+
+        for ((file, dir), info) in self.files.iter() {
+            w.write(file)?;
+            w.write_u8(0)?;
+            w.write_uleb128(dir.0 as u64)?;
+            w.write_uleb128(info.last_modification)?;
+            w.write_uleb128(info.length)?;
+        }
+        w.write_u8(0)?;
+
+        let header_length = (w.len() - header_length_base) as u64;
+        w.write_word_at(header_length_offset, header_length, self.format.word_size())?;
+
+        for instruction in &self.instructions {
+            instruction.write(w, self.address_size)?;
+        }
+
+        let length = (w.len() - length_base) as u64;
+        w.write_initial_length_at(length_offset, length, self.format)?;
+
+        Ok(offset)
+    }
+}
+
+/// A row in the line number table that corresponds to a machine instruction.
+#[derive(Debug, Clone, Copy)]
+pub struct LineRow {
+    /// The offset of the instruction from the start address of the sequence.
+    pub address_offset: u64,
+    /// The index of an operation within a VLIW instruction.
+    ///
+    /// The index of the first operation is 0.
+    /// Set to 0 for non-VLIW instructions.
+    pub op_index: u64,
+
+    /// The source file corresponding to the instruction.
+    pub file: FileId,
+    /// The line number within the source file.
+    ///
+    /// Lines are numbered beginning at 1. Set to 0 if there is no source line.
+    pub line: u64,
+    /// The column number within the source line.
+    ///
+    /// Columns are numbered beginning at 1. Set to 0 for the "left edge" of the line.
+    pub column: u64,
+    /// An additional discriminator used to distinguish between source locations.
+    /// This value is assigned arbitrarily by the DWARF producer.
+    pub discriminator: u64,
+
+    /// Set to true if the instruction is a recommended breakpoint for a statement.
+    pub is_statement: bool,
+    /// Set to true if the instruction is the beginning of a basic block.
+    pub basic_block: bool,
+    /// Set to true if the instruction is a recommended breakpoint at the entry of a
+    /// function.
+    pub prologue_end: bool,
+    /// Set to true if the instruction is a recommended breakpoint prior to the exit of
+    /// a function.
+    pub epilogue_begin: bool,
+
+    /// The instruction set architecture of the instruction.
+    ///
+    /// Set to 0 for the default ISA. Other values are defined by the architecture ABI.
+    pub isa: u64,
+}
+
+impl LineRow {
+    /// Return the initial state as specified in the DWARF standard.
+    fn initial_state() -> Self {
+        LineRow {
+            address_offset: 0,
+            op_index: 0,
+
+            file: FileId::initial_state(),
+            line: 1,
+            column: 0,
+            discriminator: 0,
+
+            is_statement: DEFAULT_IS_STATEMENT,
+            basic_block: false,
+            prologue_end: false,
+            epilogue_begin: false,
+
+            isa: 0,
+        }
+    }
+
+    fn default(version: u16) -> Self {
```

Comment:
Nitpick: since this can't be a `Default` impl, it should probably be called `new`, just so that it isn't slightly jarring for readers who would otherwise expect a `default` method to be part of a `Default` impl.